### PR TITLE
Feature/f 129 add error and successful toast to submit button back up

### PR DESCRIPTION
**Pull request by Esoteriker**
This pull request introduces a Snackbar component to the application, which provides user feedback through notifications. The changes include adding the Snackbar component, context, and related hooks, as well as integrating the Snackbar into various parts of the application.

### Snackbar Component Integration:

* [`src/app/layout.tsx`](diffhunk://#diff-29d0a55f5ca1a9951042587f8b891bb4cd8b204f5bc855dc2468ed44cde2f5b3R8): Added the `Snackbar` component to the root layout to ensure it is available throughout the application. [[1]](diffhunk://#diff-29d0a55f5ca1a9951042587f8b891bb4cd8b204f5bc855dc2468ed44cde2f5b3R8) [[2]](diffhunk://#diff-29d0a55f5ca1a9951042587f8b891bb4cd8b204f5bc855dc2468ed44cde2f5b3R65)
* [`src/app/providers.tsx`](diffhunk://#diff-bbb6142c7cbfbe0d99f8fa0a0c04e6d3d3f44ed2e25f0a1f7064cdce48a06382R18): Introduced `SnackbarProvider` to the application providers to manage Snackbar state and context. [[1]](diffhunk://#diff-bbb6142c7cbfbe0d99f8fa0a0c04e6d3d3f44ed2e25f0a1f7064cdce48a06382R18) [[2]](diffhunk://#diff-bbb6142c7cbfbe0d99f8fa0a0c04e6d3d3f44ed2e25f0a1f7064cdce48a06382L36-R39)

### Snackbar Usage in Components:

* [`src/components/DownloadCountryAccordions/DownloadCountryAccordions.tsx`](diffhunk://#diff-e5147fdadb2749adf08aa4b12d7501565c384f852d8f7cdf608adfdfd2b3b404R9-R19): Integrated `useSnackbar` hook to display success and error messages during the download process. [[1]](diffhunk://#diff-e5147fdadb2749adf08aa4b12d7501565c384f852d8f7cdf608adfdfd2b3b404R9-R19) [[2]](diffhunk://#diff-e5147fdadb2749adf08aa4b12d7501565c384f852d8f7cdf608adfdfd2b3b404R41-R42) [[3]](diffhunk://#diff-e5147fdadb2749adf08aa4b12d7501565c384f852d8f7cdf608adfdfd2b3b404L60-R86)
* [`src/components/Subscribe/Subscribe.tsx`](diffhunk://#diff-d64c83b8d1a8e7f50321ad6fe2af658780503b38a53524e20c51311c66de0a12R19-R22): Integrated `useSnackbar` hook to show notifications for subscription success and failure. [[1]](diffhunk://#diff-d64c83b8d1a8e7f50321ad6fe2af658780503b38a53524e20c51311c66de0a12R19-R22) [[2]](diffhunk://#diff-d64c83b8d1a8e7f50321ad6fe2af658780503b38a53524e20c51311c66de0a12R47-R48) [[3]](diffhunk://#diff-d64c83b8d1a8e7f50321ad6fe2af658780503b38a53524e20c51311c66de0a12L105-R125) [[4]](diffhunk://#diff-d64c83b8d1a8e7f50321ad6fe2af658780503b38a53524e20c51311c66de0a12L173-L190)

### Snackbar Component and Context:

* [`src/components/Snackbar/Snackbar.tsx`](diffhunk://#diff-82d214a193886b0185e2882f16e2a6ff0c316f8cc202c7f3484bb6d5eddbdf70R1-R75): Created the `Snackbar` component using `framer-motion` for animations and various icons for status indication.
* [`src/domain/contexts/SnackbarContext.tsx`](diffhunk://#diff-bd25d873928f6747414a2108d79771345efe5212c9dba1230755fc1f3bdfd23bR1-R78): Implemented the `SnackbarContext` and `SnackbarProvider` to manage Snackbar state and provide `showSnackBar` and `closeSnackBar` functions.

### Supporting Changes:

* [`src/domain/entities/download/Country.ts`](diffhunk://#diff-f8b62ef95cd91ae584b63fb322517330d44c54fd7d766478d387fbe806882c7bR42-R45): Added constants for download success and error messages.
* [`src/domain/entities/snackbar/Snackbar.ts`](diffhunk://#diff-2cdddfec15aa099e9f58031eb24a9f21543dbf87f3e63931df3830d043a21f29R1-R13): Defined default Snackbar properties and durations.
* [`src/domain/enums/Snackbar.ts`](diffhunk://#diff-78c8461373ae5fae308a4fcd6acdf9cec25ea3356115a5f91371f3e32c4ad2b5R1-R18): Created enums for Snackbar status and position.
* [`src/domain/props/SnackbarProps.tsx`](diffhunk://#diff-029019e69da48c5720760385a1d4e65e003e8efb98ac7318fb68007c0fd486eaR1-R47): Added utility functions to get CSS classes based on Snackbar position and status, and defined the `SnackbarProps` interface.

-----------------------------------------------------------------------------------------------------

main UI effect, remove submit button status and icon, instead of that use snackbar to show if a form submit is success or error, now we can use snackbar as long as needed, to have more method to inform user a event state

### CODE REVIEW

File: src/components/Snackbar/Snackbar.tsx [hunk 1]
```diff
@@ -0,0 +1,75 @@
...
+        return <TickCircle size={24} className="text-white" />;
+      case SnackbarStatus.Warning:
+        return <TriangleAlert size={24} className="text-white" />;
+      case SnackbarStatus.Error:
+        return <InfoCircle size={24} className="text-white" />;
+      case SnackbarStatus.Default:
+      default:
+        return null;
+    }
+  };
```
Comment (marinovl7) [inline src/components/Snackbar/Snackbar.tsx:52]: Export this into operations file

**Review by marinovl7 — CHANGES_REQUESTED** [review]
Great job, overall. Can you please use the error messages that the backend team provides when the subscription didn't work? If there is something else like a netork issue or something like that?

**Comment by marinovl7** [comment]
Also, could you please display the snackbar messgae bottom right?

**Comment by bohdangarchu** [comment]
Why can't I select topics in this version?

Uploading Screen Recording 2024-12-08 at 12.27.57.mov…

**Comment by Esoteriker** [comment]
> Why can't I select topics in this version?
> 
>  Screen.Recording.2024-12-08.at.12.27.57.mov



> Why can't I select topics in this version?
> 
>  Screen.Recording.2024-12-08.at.12.27.57.mov

**Comment by bohdangarchu** [comment]
> Why can't I select topics in this version?
> 
>  Screen.Recording.2024-12-08.at.12.27.57.mov

resolved

**Review by bohdangarchu — APPROVED** [review]

**Pull request merged**